<?xml version="1.0" encoding="UTF-8"?>
<launch>

  <arg name="rviz"  default="true"/>

  <!-- Load definition of pregrasp motion into play_motion namespace -->
  <rosparam command="load" file="$(find tiago_dual_pick_place)/config/pick_motions.yaml" /> 

  <!-- Pick & place server -->
  <node name="pick_place_server" pkg="tiago_dual_pick_place" type="pick_place_server.py" output="screen">
      <rosparam command="load" file="$(find tiago_dual_pick_place)/config/pick_place_params.yaml" />
      <param name="move_group_l0"  value="arm_left" />
      <param name="move_group_l1"  value="arm_left_torso" />
      <param name="grasp_frame_l"  value="arm_left_tool_link" />
      <param name="move_group_r0"  value="arm_right" />
      <param name="move_group_r1"  value="arm_right_torso" />
      <param name="grasp_frame_r"  value="arm_right_tool_link" />
      <param name="object_width"  value="0.05" />
      <param name="object_height" value="0.10" />
      <param name="object_depth"  value="0.05" />
  </node>

  <!-- Node exposing service to start looking for the object and trigger the picking -->
  <node name="pick_client" pkg="tiago_dual_pick_place" type="pick_place_client.py" output="screen"/>  

  <group if="$(arg rviz)">
      <node name="pick_place_rviz" pkg="rviz" type="rviz" args="-d $(find tiago_dual_pick_place)/config/rviz/tiago_dual_pick_place.rviz" />
  </group>

</launch>
